src/Features/Core/Portable/ImplementInterface/AbstractImplementInterfaceService.CodeAction.cs

```diff
@@ -407,9 +407,20 @@ private static bool IsUnexpressibleTypeParameter(ITypeParameterSymbol typeParame
 
             private SyntaxNode CreateThroughExpression(SyntaxGenerator factory)
             {
-                var through = ThroughMember.IsStatic
-                    ? factory.IdentifierName(State.ClassOrStructType.Name)
```

Comment:
i'd prefer you kept the single ternary.  And just updated the true-branch  (to call a helper) that gives you the static "through" name.